packages/compartment-mapper/src/link.js

```diff
@@ -391,13 +441,21 @@ export const link = (
       modules,
       scopes,
       exitModules,
+      attenuators,
       archiveOnly,
     );
     const resolveHook = resolve;
     resolvers[compartmentName] = resolve;
 
-    // TODO also thread powers selectively.
-    const compartment = new Compartment(globals, undefined, {
+    let personalGlobals = Object.create(null);
+    if (!archiveOnly) {
+      personalGlobals = getAllowedGlobals(
```

Comment:
If the person in question is the compartment, consider naming this `compartmentGlobals`.